### PR TITLE
SONAR-9099 return visibility for views + fix quality flaws

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [project](https://github.com/SonarSource/sonarqube/blob/a60862b473c64191bba19897d56ed83026a543af/server/sonar-server/src/main/java/org/sonar/server/project): Add a 'package-info.java' file to document the 'project' package [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml [hunk 1]
```diff
@@ -181,14 +181,14 @@
   </sql>
 
   <select id="selectDefaultGroupIdByUuid" resultType="Integer">
-    select org.default_group_id as "defaultGroupId"
+    select org.default_group_id
```
Comment (sns-seb) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml]: only this mapping is useless, as the result is mapped to an Integer type

File: server/sonar-db-dao/src/main/resources/org/sonar/db/component/ComponentMapper.xml [hunk 2]
```diff
@@ -340,7 +340,7 @@
   </sql>
 
   <select id="selectUuidsForQualifiers" resultType="UuidWithProjectUuid">
-    SELECT p.uuid as "uuid", p.project_uuid as "projectUuid" FROM projects p
+    SELECT p.uuid, p.project_uuid as "projectUuid" FROM projects p
```
Comment (sns-seb) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/component/ComponentMapper.xml]: restore this mapping, as the result is mapped to a Bean. Mapping is used to map to a bean of the property. It should be kept.

File: server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml [hunk 3]
```diff
@@ -181,14 +181,14 @@
...
     from organizations org
     where
     org.uuid = #{uuid, jdbcType=VARCHAR}
   </select>
 
   <select id="selectNewProjectPrivateByUuid" resultType="Boolean">
-    select org.new_project_private as "newProjectPrivate"
+    select org.new_project_private
```
Comment (sns-seb) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml]: :+1:  this mapping also is useless

File: server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml [hunk 4]
```diff
@@ -3,13 +3,13 @@
 
 <mapper namespace="org.sonar.db.organization.OrganizationMapper">
   <sql id="selectColumns">
-    org.uuid as "uuid",
+    org.uuid,
```
Comment (sns-seb) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml]: restore the mappings in "selectColumns"

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+  private final boolean isPrivate;
+  private final String word;
+
+  Visibility(boolean isPrivate, String word) {
+    this.isPrivate = isPrivate;
+    this.word = word;
+  }
+
+  public String asWord() {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: Rename `getLabel` as it's done in other enum

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    return word;
+  }
+
+  boolean asBoolean() {
+    return isPrivate;
+  }
+
+  public static String asWord(boolean isPrivate) {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: the stream seems overkill here. `return isPrivate ? PRIVATE.getLabel() : PUBLIC.getLabel()`

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+  public static boolean asBoolean(String word) {
+    return stream(values())
+      .filter(v -> v.asWord().equals(word))
+      .map(Visibility::asBoolean)
+      .findAny()
+      .orElseThrow(() -> new IllegalStateException("Invalid visibility word '" + word + "'"));
+  }
+
+  public static String[] words() {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: why an array and not a list?

File: server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml (hunk 4 above)
Comment (bartfastiel) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml]: Done.

File: server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml]: Ok.

File: server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml (hunk 3 above)
Comment (bartfastiel) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/organization/OrganizationMapper.xml]: Ok.

File: server/sonar-db-dao/src/main/resources/org/sonar/db/component/ComponentMapper.xml (hunk 2 above)
Comment (bartfastiel) [inline server/sonar-db-dao/src/main/resources/org/sonar/db/component/ComponentMapper.xml]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java (hunk 6 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: As suggested by @teryk I'll use `getLabel`.

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java (hunk 7 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: I changed it to a `Set`

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java (hunk 5 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+  public static String asWord(boolean isPrivate) {
+    return stream(values())
+      .filter(v -> v.asBoolean() == isPrivate)
+      .map(Visibility::asWord)
+      .findAny()
+      .orElseThrow(() -> new IllegalStateException("Invalid visibility boolean '" + isPrivate + "'"));
+  }
+
+  public static boolean asBoolean(String word) {
```
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: I changed it to `isPrivate`

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    return stream(values())
+      .filter(v -> v.asWord().equals(word))
+      .map(Visibility::asBoolean)
+      .findAny()
+      .orElseThrow(() -> new IllegalStateException("Invalid visibility word '" + word + "'"));
+  }
+
+  public static String[] words() {
+    return stream(values()).map(Visibility::asWord).toArray(String[]::new);
```
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: Done. 👍

File: server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java (hunk 6 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/Visibility.java]: I agree that it is tending to be over engineered 👍 , but I still want to keep it for these reasons:
* To make the code consistent with the `isPrivate` method
* To enforce consistency with the enum constant configuration (we also have defined the booleans "true" and "false" there, I do not want to hardcode it down here again)

With @sns-seb I also discussed the possibility to convert the stream to a loop, but that seems not to add value.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [project](https://github.com/SonarSource/sonarqube/blob/bbfb24bc0021784ead269e83b24aab90a6e601da/server/sonar-server/src/main/java/org/sonar/server/project): Add a 'package-info.java' file to document the 'project' package [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**